### ISSUE TITLE
Add an option for SPA

### ISSUE THREAD

**Issue by eruvanos** [pyscript/pyscript-cli#132]
For single page applications it would be great to have a spa modus, which serves a default (often just index.html) page in case there is no matching file within the folder.

Example (old Python):
https://gist.github.com/jtangelder/e445e9a7f5e31c220be6

**Comment by FabioRosado** [comment]
Hello I am going through the open issues and triaging them before we do another release of the CLI.

Currently (in main) we have a `pyscript run` which runs a web server in a folder and serves whatever is there, even if index.html isn't available. 

Does this do what you expect or you mean something else?

**Comment by eruvanos** [comment]
Hi,
to serve some SPA setups, it is required, to always serve index.html for any route, except for those which actually refer to an existing file to serve.

This way the routing can use path based routing ("/<app-route>") instead of "#/<app-route>".

**Comment by FabioRosado** [comment]
What exactly would you propose we change in the CLI for this?

**Comment by eruvanos** [comment]
Maybe a `--default <file>.html` parameter would be a good option, which always returns the given file if the requested path does not exist.

So it would be flexible and would not be tied down to the SPA use case.

**Comment by eruvanos** [comment]
No, it requires to always answer with the index file, if the path does not exist.
